app/controllers/areas_controller.rb

```diff
@@ -18,7 +18,11 @@ def index
   end
 
   def search
-    api_response = Imminence.mapit_api.location_for_postcode(params[:postcode])
+    # Strip trailing whitespace, most non-alphanumerics, and use the
+    # uk_postcode gem to potentially transpose O/0 and I/1.
+    sanitized_postcode = UKPostcode.parse(params[:postcode].gsub(/[^\w\s]/i, '').strip).to_s
```

Comment:
I thought about making it into a `before`, or a separate method, but in the end I took the same approach as in https://github.com/alphagov/business-support-finder/pull/94. As discussed in person, I'll look into it though.